File: plugins/hello.c
#include "inari.h"
#include "plugin.h"
#include "command.h"

void cmd_plugin_hello(message_t);

void plugin_init(command_handle_t* handle) {
  handle->name = "plugin";
  handle->fcn = cmd_plugin_hello;
}

void cmd_plugin_hello(message_t msg) {
  irc_privmsgf(*msg.irc, msg.chan, "%s: Hello from an external plugin!", msg.nick);
}

File: .inari/lua/hello.lua
--[[
-- This is a simple lua plugin designed to test out
-- inari's Lua/C bridge
--]]

PluginName = "lua"
AdminOnly  = false

function plugin_init ()
  -- nothing to be done for init
end

function plugin (irc)
  irc:privmsg(irc.chan, 2, irc.nick .. ": Hello from lua!")
end
